File: CWE_JavaScipt/src/CWE416/Complaint.js
//      CWE-416	Use After Free prevention


function createUser() {
  let user = { name: "John", email: "john@example.com" };
  return user;
}

let user = createUser();
console.log(user.name);
user = null;
if (!user) {
    console.log("User is Null");

} else {
  console.log(user.name);
}

File: CWE_Java/src/main/java/com/example/cwe/CWE416/Complaint.java
package com.example.cwe.CWE416;

//      CWE-416: Use After Free

public class Complaint {
    public static void main(String[] args) {

        Complaint obj = new Complaint();
        obj.getStatus();
        obj = null;
        System.gc();

        if(obj!=null)
        {
            obj.getStatus();
        }

        System.out.println("Object is null");   // obj is used after it is free

    }

    void getStatus(){
        System.out.println("Method of Complaint Class");
    }

}